Virtual reality componentization.

Ghommie opened the pull request:
## About The Pull Request
Make virtual reality less hardcoded.

## Why It's Good For The Game
Bugfixing. Will close #8520.

## Changelog
:cl:
refactor: turned virtual reality into a component datum, which is then applied to spawned virtual mobs.
This fixes mob transformations (such as wabbajack and monkeyizing) breaking the previously hardcoded behaviour and trapping you in VR, also enabling a more concrete virtual reality inception experience.
/:cl:

Ghommie committed:
componentization part 1: "The flawed theorical code"

Ghommie committed:
part 1.1

deathride58 commented on code/datums/diseases/transformation.dm:
```diff
@@ -63,11 +63,11 @@
-			else
-				new_mob.key = affected_mob.key
+				new_mob.a_intent = INTENT_HARM
+				if(affected_mob.mind)
```
????
whis is this being placed into the jobban check?

Ghommie committed:
wrong indention.

Ghommie committed:
*Dramatic low-pitch scenic sound plays in*

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
Purging old code,
forcing mind transfer on transform procs,
Fixing a slight fuck up null.ckey :eyes:

Ghommie committed:
Progress.

Ghommie committed:
fml

Ghommie committed:
The big part is done.

Ghommie committed:
Moving the snowflake mob var to the component and finalizing.

Ghommie committed:
pet peeve.

Ghommie committed:
Recompiling TGUI.

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
key, ckey, key, ckey

Ghommie committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into Ghommie-cit45

Ghommie committed:
unrandomizes mind names.

Ghommie committed:
Fixing head rev blebs
